src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -667,13 +667,12 @@ class VenvScriptWriter:
 
     @classmethod
     def create(
-        cls, pex_environment: PexEnvironment, pex: Pex, venv_rel_dir: PurePath
+        cls, complete_pex_env: CompletePexEnvironment, pex: Pex, venv_rel_dir: PurePath
     ) -> VenvScriptWriter:
         # N.B.: We don't know the working directory that will be used in any given
         # invocation of the venv scripts; so we deal with working_directory once in an
         # `adjust_relative_paths` function inside the script to save rule authors from having to do
         # CWD offset math in every rule for all the relative paths their process depends on.
-        complete_pex_env = pex_environment.in_sandbox(working_directory=None)
```

Comment:
It looks like you stripped the comment of its claim. The now `CompletePexEnvironment` may have a known working directory, which seems to suggest an assert complete_pex_env._working_directory is None, but that's obviously begging why let people pass this in at all?

Reply:
So I'm only partially grokking the comment, but it still feels true, albeit missing a qualifier.

Should it not start with "We _may_ not know the ..."?

Reply:
Oh and "why let people pass this in at all?": Because in the case of `run`, `working_directory` very much is set to something: the build root!

Reply:
Aha, missed that obvious case given the PR subject. As to the comment, it seems like "in any given invocation" -> "in any given sandbox invocation" then makes it True again with the creation of the `complete_pex_env` now delegated upstream. Not worth another CI burn though.